### PR TITLE
External Contracts as Byte Code rather than Source Code

### PR DESCRIPTION
**Pull request by BrendanChou**

### CODE REVIEW

File: migrations/2_deploy.js [hunk 1]
```diff
@@ -55,30 +53,73 @@ const TokenA = artifacts.require("TokenA");
...
 
-function maybeDeployTestTokens(deployer, network) {
+// Helper functions
+function parseExternalContracts(contracts) {
+  const classDefaults = {
+    from: web3.eth.accounts[0],
+    gas: 6721975,
+    gasPrice: 100000000000
+  };
+  for (let i in contracts) {
```
Comment (AntonioJuliano) [inline migrations/2_deploy.js]: does this work? won't this give you the items themselves and not the indexes?

File: migrations/2_deploy.js [hunk 2]
```diff
@@ -55,30 +53,73 @@ const TokenA = artifacts.require("TokenA");
...
 // Other constants
 const BigNumber = require('bignumber.js');
 const ONE_HOUR = new BigNumber(60 * 60);
 
-function maybeDeployTestTokens(deployer, network) {
+// Helper functions
+function parseExternalContracts(contracts) {
+  const classDefaults = {
+    from: web3.eth.accounts[0],
+    gas: 6721975,
```
Comment (AntonioJuliano) [inline migrations/2_deploy.js]: can you load these from truffle.js based on network?

File: migrations/2_deploy.js [hunk 3]
```diff
@@ -55,30 +53,73 @@ const TokenA = artifacts.require("TokenA");
...
+    gasPrice: 100000000000
+  };
+  for (let i in contracts) {
+    let contract = contracts[i];
+    contract.setProvider(web3.currentProvider);
+    contract.setNetwork(web3.currentNetwork);
```
Comment (AntonioJuliano) [inline migrations/2_deploy.js]: use the network that's passed in to `doMigration` not the one on web3

File: migrations/2_deploy.js [hunk 4]
```diff
@@ -55,30 +53,73 @@ const TokenA = artifacts.require("TokenA");
...
 
-function maybeDeploy0x(deployer, network) {
+async function maybeDeploy0xV2(deployer, network) {
   if (isDevNetwork(network)) {
-    return deployer.deploy(ZeroExProxyV1)
-      .then(() => deployer.deploy(ZeroExExchangeV1, FeeToken.address, ZeroExProxyV1.address))
-      .then(() => ZeroExProxyV1.deployed())
-      .then(proxy => proxy.addAuthorizedAddress(ZeroExExchangeV1.address));
+    parseExternalContracts([ZeroExExchangeV2, ZeroExProxyV2]);
+    const zrxAssetData = "0xf47261b0000000000000000000000000" + FeeToken.address.substr(2);
```
Comment (AntonioJuliano) [inline migrations/2_deploy.js]: I think you should not do this kind of stuff manually but should instead use the functions provided by 0x.js. If you need to install multiple versions of 0x.js see https://github.com/dydxprotocol/orderbook/blob/master/package.json#L33

File: migrations/2_deploy.js (hunk 1 above)
Comment (BrendanChou) [reply migrations/2_deploy.js]: javascript gives the indexes

**Pull request merged**